### ISSUE TITLE
[help] v4VendorSpecificOption in config file doesn't work

### ISSUE THREAD

**Issue by YbAndrew** [jagornet/dhcp#68]
I need to response suboption in v4VendorSpecificOption  to client. 
my config xml is:
```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<ns2:dhcpServerConfig xmlns:ns2="http://jagornet.com/dhcp/xml">
    <v4ServerIdOption>
        <ipAddress>10.17.0.14</ipAddress>
    </v4ServerIdOption>
    <v6ServerIdOption>
        <opaqueData>
            <hexValue>000100016038A683D60582BAF453</hexValue>
        </opaqueData>
    </v6ServerIdOption>
    <policies>
        <policy>
            <name>dhcp.ignoreLinkLocal</name>
            <value>false</value>
             <name>dhcp.ignoreSelfPackets</name>
            <value>false</value>
        </policy>
    </policies>
    <links>
        <link>
            <name>IPv4 Client Link 1</name>
            <address>10.17.0.0/24</address>
            <policies>
                <policy>
                    <name>v4.defaultLeasetime</name>
                    <value>0</value>
                </policy>
            </policies>
            <v4ConfigOptions>
                <v4SubnetMaskOption>
                    <ipAddress>255.255.255.0</ipAddress>
                </v4SubnetMaskOption>
                <v4RoutersOption>
                    <ipAddress>10.17.0.1</ipAddress>
                </v4RoutersOption>
        <v4VendorSpecificOption>
        <opaqueData>
        <subOptions>
        <optionDef v4="true" code="1" name="VendorSubopt1">
        <stringOption>
        <string>VendorSpecial</string>
       </stringOption>
        </optionDef>
        </subOptions>
        </opaqueData>
        </v4VendorSpecificOption>
            </v4ConfigOptions>
         <v4AddrPools>
                <pool>
                   <range>10.17.0.200-10.17.0.220</range>
                </pool>
          </v4AddrPools>
        </link>
    </links>
</ns2:dhcpServerConfig>
```
when sever get a request ,log is:
[epollEventLoopGroup-9-1] WARN  channel.DefaultChannelPipeline - An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.lang.NullPointerException: null
        at com.jagornet.dhcp.core.option.base.BaseOpaqueData.getLength(BaseOpaqueData.java:65) ~[jagornet-dhcp-core-3.0.3.jar:?]
        at com.jagornet.dhcp.core.option.base.BaseOpaqueDataOption.getLength(BaseOpaqueDataOption.java:69) ~[jagornet-dhcp-core-3.0.3.jar:?]
        at com.jagornet.dhcp.core.option.base.BaseDhcpOption.encodeCodeAndLength(BaseDhcpOption.java:66) ~[jagornet-dhcp-core-3.0.3.jar:?]
        at com.jagornet.dhcp.core.option.base.BaseOpaqueDataOption.encode(BaseOpaqueDataOption.java:75) ~[jagornet-dhcp-core-3.0.3.jar:?]
        at com.jagornet.dhcp.server.db.DhcpOption.fromConfigDhcpOption(DhcpOption.java:170) ~[jagornet-dhcp-server-3.0.3.jar:3.0.3]
        at com.jagornet.dhcp.server.request.binding.BaseBindingManager.convertDhcpOptions(BaseBindingManager.java:1000) ~[jagornet-dhcp-server-3.0.3.jar:3.0.3]
        at com.jagornet.dhcp.server.request.binding.BaseBindingManager.createBinding(BaseBindingManager.java:399) ~[jagornet-dhcp-server-3.0.3.jar:3.0.3]
        at com.jagornet.dhcp.server.request.binding.V4AddrBindingManagerImpl.createDiscoverBinding(V4AddrBindingManagerImpl.java:270) ~[jagornet-dhcp-server-3.0.3.jar:3.0.3]
        at com.jagornet.dhcp.server.request.DhcpV4DiscoverProcessor.process(DhcpV4DiscoverProcessor.java:112) ~[jagornet-dhcp-server-3.0.3.jar:3.0.3]
        at com.jagornet.dhcp.server.request.BaseDhcpV4Processor.processMessage(BaseDhcpV4Processor.java:221) ~[jagornet-dhcp-server-3.0.3.jar:3.0.3]
        at com.jagornet.dhcp.server.request.DhcpV4MessageHandler.handleMessage(DhcpV4MessageHandler.java:90) ~[jagornet-dhcp-server-3.0.3.jar:3.0.3]
        at com.jagornet.dhcp.server.netty.DhcpV4ChannelHandler.channelRead0(DhcpV4ChannelHandler.java:77) ~[jagornet-dhcp-server-3.0.3.jar:3.0.3]
        at com.jagornet.dhcp.server.netty.DhcpV4ChannelHandler.channelRead0(DhcpV4ChannelHandler.java:52) ~[jagornet-dhcp-server-3.0.3.jar:3.0.3]
        at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99) ~[netty-all-4.1.48.Final.jar:4.1.48.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-all-4.1.48.Final.jar:4.1.48.Final]
        at io.netty.channel.AbstractChannelHandlerContext.access$600(AbstractChannelHandlerContext.java:61) ~[netty-all-4.1.48.Final.jar:4.1.48.Final]
        at io.netty.channel.AbstractChannelHandlerContext$7.run(AbstractChannelHandlerContext.java:370) ~[netty-all-4.1.48.Final.jar:4.1.48.Final]
        at io.netty.util.concurrent.DefaultEventExecutor.run(DefaultEventExecutor.java:66) ~[netty-all-4.1.48.Final.jar:4.1.48.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) [netty-all-4.1.48.Final.jar:4.1.48.Final]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-all-4.1.48.Final.jar:4.1.48.Final]
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-all-4.1.48.Final.jar:4.1.48.Final]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_171]
# how to fix it? Thanks!

**Comment by YbAndrew** [comment]
i find this funcation in DhcpV4ConfigOptions
```
if (configOptions.getV4VendorSpecificOption() != null) {
			V4VendorSpecificOption vendorSpecificOption = configOptions.getV4VendorSpecificOption();
			if (vendorSpecificOption != null) {
				optionMap.put((int)vendorSpecificOption.getCode(),
						new DhcpV4VendorSpecificOption(
								OpaqueDataUtil.toBaseOpaqueData(
										vendorSpecificOption.getOpaqueData())));
			}
		}
```
and thi funcation in OpaqueDataUtil
```
public static BaseOpaqueData toBaseOpaqueData(OpaqueData opaque) {
    	BaseOpaqueData baseOpaque = new BaseOpaqueData();
    	baseOpaque.setAscii(opaque.getAsciiValue());
    	baseOpaque.setHex(opaque.getHexValue());
    	return baseOpaque;
    }
```
It seem that suboption is not support in V4VendorSpecificOption

**Comment by agrabil** [comment]
Fixed by #51
